### PR TITLE
Migration over to the new ViewCoordinates

### PR DESCRIPTION
**Pull request by jleibs**
### What
Remove the legacy component.
Clean up viewer usages.
Migrate backing python APIs to the new API.
Introduce examples for all languages.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3398) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3398)
- [Docs preview](https://rerun.io/preview/97ed81cbff94ebd7a6dce38f0c5e7dda7017cd69/docs) 
- [Examples preview](https://rerun.io/preview/97ed81cbff94ebd7a6dce38f0c5e7dda7017cd69/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_space_view/src/unreachable_transform_reason.rs [hunk 1]
```diff
@@ -13,6 +13,9 @@ pub enum UnreachableTransformReason {
 
     /// Unknown transform between this and the reference space.
     DisconnectedSpace,
+
+    /// View coordinates contained an invalid value
+    InvaliViewCoordinates,
```
Comment (Wumpf) [inline crates/re_space_view/src/unreachable_transform_reason.rs:18]:
```suggestion
    InvalidViewCoordinates,
```

File: crates/re_types/definitions/rerun/archetypes/view_coordinates.fbs [hunk 2]
```diff
@@ -7,6 +7,8 @@ namespace rerun.archetypes;
 
 
 /// How we interpret the coordinate system of an entity/space.
+///
+/// \example view_coordinates_simple
```
Comment (Wumpf) [inline crates/re_types/definitions/rerun/archetypes/view_coordinates.fbs:11]: We should leave a not what we do if no ViewCoordinates is present, i.e. how we interpret the coordinate system of a space in absence of explicit specification

File: docs/code-examples/view_coordinates_simple.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Log a batch of 3D arrows.
```
Comment (Wumpf) [inline docs/code-examples/view_coordinates_simple.cpp:1]:
```suggestion
// Change the view coordinates for the scene.
```

File: docs/code-examples/view_coordinates_simple.rs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+//! Log a batch of 3D arrows.
```
Comment (Wumpf) [inline docs/code-examples/view_coordinates_simple.rs:1]:
```suggestion
//! Change the view coordinates for the scene.
```

File: docs/code-examples/view_coordinates_simple.rs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let (rec, storage) = RecordingStreamBuilder::new("rerun_example_view_coordinate").memory()?;
+
+    rec.log("/", &ViewCoordinates::ULB)?;
+    rec.log(
+        "xyz",
+        &Arrows3D::new(vec![[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]) // vectors
+            .with_colors(vec![[255, 0, 0], [0, 255, 0], [0, 0, 255]]), // colors
```
Comment (Wumpf) [inline docs/code-examples/view_coordinates_simple.rs:15]: nit: this is the only one where it says "vectors" and "colors". Not sure that adds value

Comment (teh-cmc) [inline docs/code-examples/view_coordinates_simple.rs:15]:
```suggestion
        &Arrows3D::new([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]) // vectors
            .with_colors([[255, 0, 0], [0, 255, 0], [0, 0, 255]]), // colors

File: docs/code-examples/view_coordinates_simple.cpp [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+#include <rerun.hpp>
+
+#include <cmath>
+#include <numeric>
+
+namespace rr = rerun;
+
+int main() {
+    auto rec = rr::RecordingStream("rerun_example_view_coordinate");
```
Comment (Wumpf) [inline docs/code-examples/view_coordinates_simple.cpp:11]: nit:
```suggestion
    auto rec = rr::RecordingStream("rerun_example_view_coordinates");
```

File: rerun_py/rerun_sdk/rerun/log_deprecated/transform.py [hunk 7]
```diff
@@ -103,30 +117,34 @@ def log_view_coordinates(
...
-            up,
-            right_handed,
-            timeless,
-            recording=recording,
-        )
+
+        if (up, right_handed) not in _up_attrs:
+            raise ValueError("Invalid up value.")
+
+        up = _up_attrs[(up, right_handed)]
```
Comment (Wumpf) [inline rerun_py/rerun_sdk/rerun/log_deprecated/transform.py:142]: should we validate the `up` argument first for a more clean error message if it's an invalid string?

**Review by Wumpf — APPROVED** [review]
lgtm except a few nits and comment suggestions.
Didn't realize this was one was still bridg-implemented :)

**Review by Wumpf — APPROVED** [review]

**Pull request merged**